velero-plugin-for-aws/object_store.go

```diff
@@ -218,6 +234,34 @@ func (o *ObjectStore) Init(config map[string]string) error {
 	return nil
 }
 
+func readCustomerKey(customerKeyEncryptionFile string) (*string, error) {
+	if _, err := os.Stat(customerKeyEncryptionFile); err != nil {
+		if os.IsNotExist(err) {
+			return nil, errors.Wrapf(err, "provided %s does not exist: %s", customerKeyEncryptionFileKey, customerKeyEncryptionFile)
+		}
+		return nil, errors.Wrapf(err, "could not stat %s: %s", customerKeyEncryptionFileKey, customerKeyEncryptionFile)
+	}
+
+	fileHandle, err := os.Open(customerKeyEncryptionFile)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not read %s: %s", customerKeyEncryptionFileKey, customerKeyEncryptionFile)
+	}
+
+	keyBytes := make([]byte, 32)
+	nBytes, err := fileHandle.Read(keyBytes)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not read %s: %s", customerKeyEncryptionFileKey, customerKeyEncryptionFile)
+	}
+	fileHandle.Close()
+
+	if nBytes != 32 {
+		return nil, errors.Wrapf(err, "contents of %s (%s) are not exactly 32 bytes", customerKeyEncryptionFileKey, customerKeyEncryptionFile)
+	}
+
+	key := string(keyBytes)
+	return &key, nil
```

Comment:
Any reason for using the pointer type?

Reply:
I'd prefer using an empty string instead of `nil` to avoid string <-> pointer converts.

Reply:
Fixed